src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -1001,30 +987,87 @@ private static PlatformAttributes CopyAllAttributes(PlatformAttributes copyTo, P
             {
-                bool added = false;
+                SmallDictionary<string, PlatformAttributes>? childAttributes = null;
                 foreach (AttributeData attribute in immediateAttributes)
                 {
                     if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
                     {
-                        added |= TryAddValidAttribute(ref attributes, attribute);
+                        TryAddValidAttribute(ref childAttributes, attribute);
+                    }
+                }
+
+                if (childAttributes != null)
+                {
+                    if (parentAttributes != null && parentAttributes.Any())
+                    {
+                        foreach (var (platform, attributes) in parentAttributes)
+                        {
+                            if (DenyList(attributes) &&
+                                !parentAttributes.Any(ca => AllowList(ca.Value)))
+                            {
+                                // if all are deny list then we can add the child attributes
+                                foreach (var (name, childAttribute) in childAttributes)
+                                {
+                                    if (parentAttributes.TryGetValue(name, out var existing))
+                                    {
+                                        // but don't override existing unless narrowing the support
+                                        if (childAttribute.UnsupportedFirst != null &&
+                                            childAttribute.UnsupportedFirst < attributes.UnsupportedFirst)
+                                        {
+                                            attributes.UnsupportedFirst = childAttribute.UnsupportedFirst;
+                                        }
+                                    }
+                                    else
+                                    {
+                                        parentAttributes[name] = childAttribute;
+                                    }
+                                }
+                                return;
+                            }
+                            else if (AllowList(attributes))
+                            {
+                                // only attributes with same platform matter, could narrow the list
+                                if (childAttributes.TryGetValue(platform, out var childAttribute))
+                                {
+                                    // Only later versions could narrow, other versions ignored 
+                                    if (childAttribute.SupportedFirst > attributes.SupportedFirst)
+                                    {
+                                        attributes.SupportedSecond = childAttribute.SupportedFirst;
+                                    }
+
+                                    if (childAttribute.UnsupportedFirst != null)
```

Comment:
Revert + return to reduce nesting.

Reply:
We cannot return on revert, need to continue to the next attribute for other platform